.github/workflows/build.yaml

```diff
@@ -92,7 +92,7 @@ jobs:
     secrets: inherit
     uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.04
     with:
-      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10" and (.CUDA_VER == "11.8.0" or .CUDA_VER == "12.2.2")))
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.11" and (.CUDA_VER == "11.8.0" or .CUDA_VER == "12.2.2")))
```

Comment:
Is there a way for us to use `max_by` filters rather than hardcode these values?

I'm not a jq expert. I don't know how to write the query `CUDA_VER == max(CUDA_11_VERSIONS) or CUDA_VER == max(CUDA_12_VERSIONS)` but that's what we need.

We've used `matrix_filter: map(select(.ARCH == "amd64")) | max_by(.CUDA_VER) | [.]` elsewhere before when we only care about one CUDA version.

Reply:
cc: @ajschmidt8 if you know how to do this

Reply:
There is, @ajschmidt8 shared some ideas about that at https://github.com/rapidsai/dask-cuda/pull/1294#discussion_r1417857509

That thread produced this still-open issue: https://github.com/rapidsai/dask-cuda/issues/1296

I didn't pursue those ideas in this PR because I didn't want to add another thing to the mix of what was being tested, in the interest of unblocking CI as soon as possible.

Reply:
Just realized that issue is in `dask-cuda`, maybe https://github.com/rapidsai/build-planning would be a better place, to look into this RAPIDS-wide?

But either way, I think we should defer that to another PR.

Reply:
Okay, I found something even better. This handles the final `max_by` step more clearly, with tie-breaking over both Python and CUDA versions. I'll file a separate PR for this.

```jq
map(select(.ARCH == "amd64")) | group_by(.CUDA_VER|split(".")|map(tonumber)|.[0]) | map(max_by([(.PY_VER|split(".")|map(tonumber)), (.CUDA_VER|split(".")|map(tonumber))]))
```

1. Take only amd64 jobs
2. Group them by CUDA major version (11 in one list, 12 in another)
3. Map over each CUDA major version, taking the maximum over Python version first and CUDA version second.

Finally, you have a list containing two matrix entries: the highest (Python, CUDA) entry for each of the two CUDA major versions.

Reply:
> Just realized that issue is in `dask-cuda`, maybe https://github.com/rapidsai/build-planning would be a better place, to look into this RAPIDS-wide?
> 
> But either way, I think we should defer that to another PR.

Yeah IIRC that issue preceded the `build-planning` repo. In any event agree this is a good idea (especially if it is a wider-spread issue)

Transferred the Dask-cuda issue to https://github.com/rapidsai/build-planning/issues/25